manim/scene/scene.py

```diff
@@ -596,83 +596,32 @@ def get_moving_and_static_mobjects(self, animations):
+        Creates _MethodAnimations from any _AnimationBuilders and updates animation
+        kwargs with kwargs passed to play().
 
         Parameters
         ----------
-        *args : Animation or method of a mobject, which is followed by that method's arguments
+        *animations : Tuple[:class:`Animation`]
+            Animations to be played.
 
-        **kwargs : any named arguments like run_time or lag_ratio.
+        **play_kwargs
+            Configuration for the call to play().
 
         Returns
         -------
-        list : list of animations with the parameters applied to them.
-        """
-        animations = []
-        state = {
-            "curr_method": None,
-            "last_method": None,
-            "method_args": [],
-        }
-
-        def compile_method(state):
-            if state["curr_method"] is None:
-                return
-            mobject = state["curr_method"].__self__
-            if state["last_method"] and state["last_method"].__self__ is mobject:
-                animations.pop()
-                # method should already have target then.
-            else:
-                mobject.generate_target()
-            #
-            if len(state["method_args"]) > 0 and isinstance(
-                state["method_args"][-1], dict
-            ):
-                method_kwargs = state["method_args"].pop()
-            else:
-                method_kwargs = {}
-            state["curr_method"].__func__(
-                mobject.target, *state["method_args"], **method_kwargs
-            )
-            animations.append(MoveToTarget(mobject))
-            state["last_method"] = state["curr_method"]
-            state["curr_method"] = None
-            state["method_args"] = []
-
-        for arg in args:
-            if isinstance(arg, Animation):
-                compile_method(state)
-                animations.append(arg)
-            elif inspect.ismethod(arg):
-                compile_method(state)
-                state["curr_method"] = arg
-            elif state["curr_method"] is not None:
-                state["method_args"].append(arg)
-            elif isinstance(arg, Mobject):
-                raise ValueError(
-                    """
-                    I think you may have invoked a method
-                    you meant to pass in as a Scene.play argument
-                    """
-                )
-            else:
-                raise ValueError("Invalid play arguments")
-        compile_method(state)
-
+        Tuple[:class:`Animation`]
+            Animations to be played.
+        """
         for animation in animations:
-            # This is where kwargs to play like run_time and rate_func
-            # get applied to all animations
-            animation.update_config(**kwargs)
-
+            if inspect.ismethod(animation):
+                raise TypeError(
+                    "Passing mobject methods to Scene.play is no longer supported. Use "
```

Comment:
Regarding the deprecation warning, considering this is an integral part of the code of many people, we should at the very least add a warning like `If you have old scripts you would like to render, consider downgrading to manim v0.1.1 or less`.

Reply:
I'd consider that to be pretty self-explanatory, but besides that we also don't want to encourage using older versions.